### PR TITLE
refactor: new API for golang

### PR DESCRIPTION
**Pull request by wenfengwang**
new API for golang

### CODE REVIEW

File: golang/client.go [hunk 1]
```diff
@@ -75,3 +73,42 @@ func (c *client) Disconnect() error {
...
+	for _, apply := range opts {
+		apply(defaultOpts)
+	}
+
+	value, ok := c.publisherCache.Load(defaultOpts.eventbus)
+	if ok {
+		return value.(Publisher)
+	}
+
+	value, ok = c.publisherCache.Load(defaultOpts.eventbus)
```
Comment (hwjiangkai) [inline golang/client.go]: why load twice？

File: golang/subscriber.go [hunk 2]
```diff
@@ -41,135 +47,180 @@ func (m *message) GetEvent() *v2.Event {
...
-	return s.options.SubscriptionID
+	return s.options.subscriptionID
 }
 
-func (s *subscribe) Subscribe(ctx context.Context) (<-chan Message, error) {
-	s.mu.Lock()
-	defer s.mu.Unlock()
-	if s.state == stateRunning {
-		return s.messagec, nil
+func (s *subscribe) Send(ctx context.Context, event *cloudevents.BatchEvent) (*emptypb.Empty, error) {
```
Comment (hwjiangkai) [inline golang/subscriber.go:138]: What is the use scenario of this method？

File: golang/example/subscribe/main.go [hunk 3]
```diff
@@ -17,43 +17,43 @@ package main
...
+		vanus.WithConsumeTimeout(time.Second),
+	)
+	defer func() {
+		_ = s.Close()
+	}()
+
+	err = s.Listen(func(ctx context.Context, msgs ...vanus.Message) error {
+		for _, msg := range msgs {
+			fmt.Printf("received a message, event: %s\n", msg.GetEvent().String())
+			_ = msg.Success()
```
Comment (hwjiangkai) [inline golang/example/subscribe/main.go]: no return value

File: golang/subscriber.go (hunk 2 above)
Comment (wenfengwang) [reply golang/subscriber.go:138]: see example: https://github.com/linkall-labs/vanus/pull/413/files#diff-1b1b5c22b325b98c8633760d2c12138cdcf4e5cdf0d76a3898cf7c50448ef3aeR149.

SDK both support Active and Inactive modes:

- Active mode: SDK builds a gRPC stream within Gateway to receive events
- Inactive mode: default mode, in this mode, SDK starts a server to listen port, and Subscription can push events directly to it.

**Review by hwjiangkai — APPROVED** [review]

**Pull request merged**